src/services/grpc/account/v2/update-pnid-permissions.ts

```diff
@@ -0,0 +1,159 @@
+import { Status, ServerError } from 'nice-grpc';
+import { getPNIDByPID } from '@/database';
+import { PNID_PERMISSION_FLAGS } from '@/types/common/permission-flags';
+import type { UpdatePNIDPermissionsRequest } from '@pretendonetwork/grpc/account/v2/update_pnid_permissions';
+import type { Empty } from '@pretendonetwork/grpc/google/protobuf/empty';
+
+export async function updatePNIDPermissions(request: UpdatePNIDPermissionsRequest): Promise<Empty> {
+	const pnid = await getPNIDByPID(request.pid);
+
+	if (!pnid) {
+		throw new ServerError(
+			Status.INVALID_ARGUMENT,
+			'No PNID found'
+		);
+	}
+
+	if (!request.permissions) {
+		throw new ServerError(
+			Status.INVALID_ARGUMENT,
+			'Permissions flags not found'
+		);
+	}
+
+	if (request.permissions.bannedAllPermanently === true) {
+		await pnid.addPermission(PNID_PERMISSION_FLAGS.BANNED_ALL_PERMANENTLY);
+	} else if (request.permissions.bannedAllPermanently === false) {
+		await pnid.clearPermission(PNID_PERMISSION_FLAGS.BANNED_ALL_PERMANENTLY);
+	}
```

Comment:
This whole section works, but I wonder if it'd be better to have a lookup map like:

```js
const permissionNameMap = {
  bannedAllPermanently: PNID_PERMISSION_FLAGS.BANNED_ALL_PERMANENTLY,
 ...
}
```

With a dynamic check for all of them:

```js
for (const key of Object.keys(permissionNameMap)) {
  if (request.permissions[key] === true) {
    await pnid.addPermission(permissionNameMap[key])
  } else if (request.permissions[key] === false) {
    await pnid.clearPermission(permissionNameMap[key]);
  }
}
```

Would save a lot of space in this file, and again make it easy to keep up to date :)

Reply:
@jonbarrow Also add to the list of things for the rework - happy for it to not be done in this PR since it's blocking